spec/requests/portfolios_spec.rb

```diff
@@ -89,17 +89,28 @@
   end
 
   describe "GET /portfolios" do
-    before do
-      get "#{api}/portfolios", :headers => default_headers
-    end
+    context "without filter" do
+      before do
+        get "#{api}/portfolios", :headers => default_headers
+      end
 
-    context 'when portfolios exist' do
-      it 'returns status code 200' do
-        expect(response).to have_http_status(200)
+      context 'when portfolios exist' do
+        it 'returns status code 200' do
+          expect(response).to have_http_status(200)
+        end
+
+        it 'returns all portfolio requests' do
+          expect(json['data'].size).to eq(1)
+        end
       end
+    end
+
+    context "with filter" do
+      it 'returns only the id specified in the filter' do
+        get "#{api}/portfolios?filter[id]=#{portfolio_id}", :headers => default_headers
 
-      it 'returns all portfolio requests' do
-        expect(json['data'].size).to eq(1)
+        expect(json["meta"]["count"]).to eq 1
```

Comment:
@lindgrenj6 Can we add a spec to do filtering based on names.
Create 2 portfolios with name test1, test2 and filter on starts_with('test')

Reply:
@syncrou Would the UI be able to take advantage of this, since this is already implemented for topology products, is the UI using this filtering there?

Reply:
I believe the UI could take advantage of it, but currently the UI is not using this filtering. @gmcculloug is the UI use a filtering a post Summit change?

Reply:
If I understand the question, the UI is using this filtering when talking to the topology api already to filter out archived `service_offerings`.

There might be other places it could be used.  cc @Hyperkid123 @lgalis

Reply:
Yes we can filter but there are still some features missing. Case sensitivity is one of them. I doubt that users fill filter with the exact character cases. Also filtering by multiple attributes is using only AND but for the UI i can imagine we would need OR in some cases.

Right now we can probably use it when filtering portfolios and portfolio items.

There is also one other unfortunate thing. With this query structure we cannot use the generated clients because if we use the `filter` parameter in the client method the generated query will look something likes this:
```
/foo/foo?filter=filterValue
```
And is something that can't be customized.